### PR TITLE
Remove Replacement Files: CivicsChooser.lua and ResearchChooser.lua

### PR DESCRIPTION
**Pull request by the-m4a**
**These changes are in the current m4a-ppe branch** -- y'all can use that branch for testing if you'd prefer to download 1 version of files instead of 5 or 6 different ones.  I'll leave the pull requests all separated, however. 

There were not a lot of differences between the base game version and the CQUI versions of these files, so both are pretty small.  
The changes in the larger files did two things:
- Hide the recommended icons (if setting was configured)
- Open the full trees instead of the side panels (if setting was configured)

**What this does**
- Replace CivicsChooser.lua and ResearchChooser.lua with _CQUI.lua files that implement only the things above
- Loads ResearchChooser.lua at 200 (with Exp2 files) to cover the base/exp1/exp2 ResearchChooser_Expansion files implemented by Firaxis (in any event we just need ours to load after theirs)

**Notes**
- ResearchChooser returns the Instance object from AddAvailableResearch, so hiding the recommended icon there is easy (the Firaxis extension files also extend AddAvailableResearch).
- CivicsChooser does not return that Instance object, so the extension is in the "View" function, where it can find those instances and manipulate the RecommendedIcon by walking the control stack.  View is the function that calls AddAvailableCivic for however many Civics are available at that moment...so hooking that function and picking through the controls it built after the fact is how it gets done... (I tried to find a more elegant way and if it exists I'd like to see it.)

**Tested**
Played a chunk of a game with EXP2.  Tested with Vanilla and EXP1.  Verified changing the settings take effect immediately.

### CODE REVIEW

**Comment by Infixo** [comment]
File names are not in line with agreed convention.

**Comment by the-m4a** [comment]
do you mean file name, referring to researchchooser_CQUI.lua?  I guessing what you meant because otherwise the files "civicschooser_CQUI.lua" and "researchchooser_CQUI.lua" match existing convention

**Comment by Infixo** [comment]
https://github.com/civfanatics/CQUI_Community-Edition/wiki/File-Replacement,-Lua-Include,-Windows-and-Linux
[...]by utilizing UpperCase filenames whenever possible, since that will keep filenames fully aligned with the casing convention utilized by the vanilla game files.
[...]we must make an exception for files that replace base game files and always make those completely lowercase.

They should be named CivicsChooser_CQUI and ResearchChooser_CQUI.

**Comment by the-m4a** [comment]
all updated.  i thought that stuff was specific to the case with NotificationPanel and the "include foo_" stuff, had misremembered.  I prefer this naming style over the all-lowercase anyway, so it's no issue with me.

File: Assets/UI/Choosers/ResearchChooser_CQUI.lua [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+-- CQUI Members
+-- ===========================================================================
+local CQUI_AlwaysOpenTechTrees = false; --Ignores events calling for this to open when true
+local CQUI_ShowTechCivicRecommendations = false;
+
+-- ===========================================================================
+function CQUI_OnSettingsUpdate()
+    -- CQUI_AlwaysOpenTechTrees is a checkbox, so it has a value of true or false
+    CQUI_AlwaysOpenTechTrees = GameConfiguration.GetValue("CQUI_AlwaysOpenTechTrees");
+    -- CQUI_ShowTechCivicRecommendations is a ComboBox, so it has a vlaue of 0 or 1
```
Comment (russegon) [inline Assets/UI/Choosers/ResearchChooser_CQUI.lua:19]: misspelling in the comment
```suggestion
    -- CQUI_ShowTechCivicRecommendations is a ComboBox, so it has a value of 0 or 1
```

File: Assets/UI/Choosers/ResearchChooser_CQUI_Basegame.lua [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-- ===========================================================================
+-- Base File
+-- ===========================================================================
+-- Note: There is no ReseachChooser_Expansion2, as ResearchChooser_Expansion1 (one) is the file that is bundled with both the Rise and Fall and Gathering Storm expansions
+include("ResearchChooser");
```
Comment (Infixo) [inline Assets/UI/Choosers/ResearchChooser_CQUI_Basegame.lua:5]: Base file is loaded twice: 1st time here and 2nd time in ResearchChooser_CQUI.lua.

**Comment by Infixo** [comment]
Proper design should be as follows.

Vanilla game.
Base: ResearchChooser
XP1: ResearchChooser_Expansion1 => ResearchChooser
XP2: ResearchChooser_Expansion1 => ResearchChooser [yes, XP2 reuses XP1 files, check out lines 365-370 in Expansion2.modinfo]

CQUI
1) We want to extend only functions from ResearchChooser
2) We want to reuse the file => it means you cannot include original ResearchChooser in it
1+2 means that you cannot include ResearchChooser in it, but instead you need 3 loaders for it.
Like this:
Base: RC_CQUI_Base => RC_CQUI + RC
XP1: RC_CQUI_XP1 => RC_CQUI + RC_Exp1 => RC
XP2: RC_CQUI_XP2 => RC_CQUI + RC_Exp1 => RC [here, can use the same file as for XP1, like FXS does]

**Comment by Infixo** [comment]
Lol, Firaxis is not following its own rules and making it extra hard. Base game's AddAvailableResearch returns an instance but XP1's version does NOT. That complicates things a bit.

**Review by Infixo — APPROVED** [review]
Instead of 2 replacement files we have 2 extension files. Pretty neat.

**Pull request merged**